src/NuGet.Protocol.Catalog/Models/ModelExtensions.cs

```diff
@@ -166,5 +166,47 @@ public static bool IsListed(this PackageDetailsCatalogLeaf leaf)
             // not present (legacy behavior).
             return leaf.Published.Year != 1900;
         }
+
+        /// <summary>
+        /// Determines if the provied package details leaf represents a SemVer 2.0.0 package. A package is considered
+        /// SemVer 2.0.0 if it's version is SemVer 2.0.0 or one of its dependency version ranges is SemVer 2.0.0.
+        /// </summary>
+        /// <param name="leaf">The catalog leaf.</param>
+        /// <returns>True if the package is SemVer 2.0.0.</returns>
+        public static bool IsSemVer2(this PackageDetailsCatalogLeaf leaf)
```

Comment:
Assume this will be used in a separate PR, for bug fix?

Reply:
Separate PR.